cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -536,6 +707,57 @@ TYPED_TEST(TestBinaryArithmeticSigned, MulOverflowRaises) {
   this->AssertBinop(Multiply, left, right, "[2, null, null]");
 }
 
+TYPED_TEST(TestBinaryArithmeticSigned, ShiftLeftLogicalOverflowRaises) {
```

Comment:
From a code organization POV, I think it would be more helpful to gather all shift tests close to each other.